src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -313,11 +313,18 @@ std::string UpdateEdgeProcessor::updateAndWriteBack(PartitionID partId,
 }
 
 
-bool UpdateEdgeProcessor::checkFilter(const PartitionID partId,
+// return code 0 pass to next step
+// -1 return ok now
+// -2 return error now
+int UpdateEdgeProcessor::checkFilter(const PartitionID partId,
                                       const cpp2::EdgeKey& edgeKey) {
     auto ret = collectEdgesProps(partId, edgeKey);
-    if (ret != kvstore::ResultCode::SUCCEEDED) {
-        return false;
+    if (insertable_ && ret == kvstore::ResultCode::ERR_KEY_NOT_FOUND) {
+        return 0;
```

Comment:
Maybe we define an enum class to represent it(0/-2/-1)  more better.

Reply:
Thanks.